build_tools/cmake/iree_macros.cmake

```diff
@@ -169,3 +207,20 @@ function(iree_add_data_dependencies)
     endif()
   endforeach()
 endfunction()
+
+# iree_add_executable_dependencies
+#
+# Adds dependency to a target in a cross-compilation-aware way. This is meant
+# to be used for dependencies that are used to generate source files and can
+# only be built on host.
```

Comment:
Same comment. Also is this only for depending on executable targets? Seems like it could be used for generic target dependencies (like on the output of iree_bytecode_module).

```suggestion
# Adds dependency on a target in a cross-compilation-aware way. This
# should be used for depending on targets that are used as part of the build,
# such as for generating files used for later build steps.
```

Reply:
It can be used for adding file dependencies given that if not cross-compiling it's just plain `add_dependencies`. If cross-compiling we have special requirements over the dependency here (with `iree_host_` prefix) which isn't likely to work out of box for file dependencies. So I prefer to keep it scoped to targets for now. Later we might want to enhance it to support other cases if that becomes useful.